python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
                                   e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
         :param core_version: Version of the core to update to. If None, the core will be updated to the latest
-                             version.
+                             version. Defaults to None.
```

Comment:
Well, if your argument is that we shouldn't have this feature because we are being lazy with the versioning of core, maybe the problem is not the pattern. :p

I see a lot of value for someone, especially in the future, of using tank core 0.x.x, to make sure they don't get a breaking change. I agree however that right now this is not necessary. However, since we're getting it for free, I decided to leave it in. Plus, we've had people who were afraid of switching minor versions. This way they could ensure they can be every minor version without ever going over.